java/src/gov/usgs/producerclient/ProducerClient.java

```
@@ -0,0 +1,313 @@
+	public static final String TIME_PER_FILE = "TimePerFile";
+	public static final String INPUT_DIRECTORY = "InputDirectory";
+	public static final String ARCHIVE_DIRECTORY = "ArchiveDirectory";
+
+	/**
+	 * Required configuration string defining the input directory
+	 */
+	private static String inputDirectory;
+
+	/**
+	 * Optional configuration string defining the archive directory
+	 */
+	private static String archiveDirectory;
+
+	/**
+	 * Required configuration string defining the input file extension
+	 */
+	private static String fileExtension;
+
+	/**
+	 * Optional configuration Long defining the number seconds before reading a
+	 * file, default is null
+	 */
+	private static Long timePerFile;
+
+	/**
+	 * Log4J logger for ProducerClient
+	 */
+	static Logger logger = Logger.getLogger(ProducerClient.class);
+
+	/**
+	 * main function for ConsumerClient
+	 *
+	 * @param args
+	 *            - A String[] containing the command line arguments.
+	 */
+	public static void main(String[] args) {
+
+		// check number of arguments
+		if (args.length == 0) {
+			System.out.println(
+					"Usage: hazdev-broker ProducerClient <configfile>");
+			System.exit(1);
+		}
+
+		// init to default values
+		inputDirectory = null;
+		archiveDirectory = null;
+		fileExtension = null;
+		timePerFile = null;
+
+		// get config file name
+		String configFileName = args[0];
+
+		// read the config file
+		File configFile = new File(configFileName);
+		BufferedReader configReader = null;
+		StringBuffer configBuffer = new StringBuffer();
+		try {
+			configReader = new BufferedReader(new FileReader(configFile));
+			String text = null;
+
+			while ((text = configReader.readLine()) != null) {
+				configBuffer.append(text).append("\n");
+			}
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		} finally {
+			try {
+				if (configReader != null) {
+					configReader.close();
+				}
+			} catch (IOException e) {
```

probably should not have an empty catch block

---

Addressed in new commit.